auth/README.md

```diff
@@ -0,0 +1,41 @@
+# Getting Started with Google Cloud Authentication
+
+<a href="https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/java-docs-samples&page=editor&open_in_editor=auth/README.md">
+<img alt="Open in Cloud Shell" src ="http://gstatic.com/cloudssh/images/open-btn.png"></a>
+
+GCP client libraries use a strategy called Application Default Credentials (ADC) to find your 
+application's credentials. When your code uses a client library, the strategy checks for your 
+credentials in the following order: First, ADC checks to see if the environment variable 
+`GOOGLE_APPLICATION_CREDENTIALS` is set. If the variable is set, ADC uses the service account file 
+that the variable points to. If the environment variable isn't set, ADC uses the default service 
+account that Compute Engine, Kubernetes Engine, App Engine, and Cloud Functions provide, for 
```

Comment:
Do you wish to mention that it fetches from the Metadata server?

Also some of this is in transition.  @jonparrott PTAL

Reply:
Oh, I just copied the most relevant paragraph from https://cloud.google.com/docs/authentication/production. I could also see what kind of comments are on the other repos' tutorials on the subject.

Reply:
Could you just link to the docs instead? This is likely to get stale. :(